### DISCUSSION TITLE
Row level security

### DISCUSSION THREAD

**Discussion by heyfromapp** [supabase/supabase#14401] [Questions]
I am not able to get this to work.

I have a profiles table with the id as a foreign key in relation to auth.users table. And has a column is_seller Boolean. Here I want to implement only users enabled as seller and authenticated has to make an entry to the products table.

session is working and I see the cookies and session info with that user id.

This the policy i have for INSERT;

```
(EXISTS ( SELECT 1
   FROM profiles
  WHERE ((profiles.id = auth.uid()) AND (profiles.is_seller = true))))
```

What am I doing wrong with this policy statement?

Any help would be appreciated.

Thanks
GJ

**Comment by GaryAustin1** [comment]
Is it a WITH CHECK policy?  If you have .select() after the insert you also need to meet your select policy.

If that is not it, you might temporarily try anon as the Role from the dropdown and TRUE as the policy to make sure you really have the session at the time of the insert.  It is a common problem not to.

**Comment by heyfromapp** [comment]
Thanks, @GaryAustin1 for the response.

Yes, the policy is created using "WITH CHECK"

I am using NextJs. When I log nextjs cookies and getUser from supabase auth - I am getting the logged-in session user info.

I have tried anon to true policy and it's giving that row-level policy error. And in the insert, i am not having any selection after that, I have the router refresh to fetch the new data on the client side.

Other attempts have the same result

```
BEGIN;
  ALTER POLICY "Only Seller can create products" ON "public"."products" WITH CHECK ((EXISTS ( SELECT 1 FROM new_profiles WHERE ((new_profiles.id = auth.uid()) AND (new_profiles.is_seller = true)))));
COMMIT;
```

```
BEGIN;
  ALTER POLICY "Only Seller can create products" ON "public"."products" WITH CHECK ((auth.uid() IN ( SELECT id FROM new_profiles WHERE (new_profiles.id = auth.uid() and new_profiles.is_seller = true))));
COMMIT;
```

```
BEGIN;
  ALTER POLICY "Only Seller can create products" ON "public"."products" WITH CHECK ((auth.uid() IN (SELECT new_profiles.id FROM new_profiles WHERE (new_profiles.is_seller = true))));
COMMIT;
```

**Comment by GaryAustin1** [comment]
Do you have a select policy that would allow the RLS select to run on the profiles table?

**Comment by heyfromapp** [comment]
Yes, I have open access to read for all.

**Comment by GaryAustin1** [comment]
I would try just authenticated role with true and see if that works.  If not try anon role with true.  If anon works then you are not really signed in when you make the call. 
Otherwise you probably will need to show tables and code, etc.  This is better done in Supabase's Discord than here for more interactive help.

**Comment by heyfromapp** [comment]
Sure, @GaryAustin1 i understand. I will connect there in Discord. 

Authenticated to True policy is working. Only when I use any expression with the select statement with other tables it's giving that error.
